### PR TITLE
Fix balloon reduction use of DAX codes

### PR DESCRIPTION
**Pull request by johnwatson484**
https://eaflood.atlassian.net/browse/SFI-1878

### CODE REVIEW

File: test/integration/local/settlement.test.js [hunk 1]
```diff
@@ -85,17 +85,6 @@ describe('update settlement status', () => {
     expect(updatedPaymentRequest.lastSettlement).toBeNull()
   })
 
-  test('should not add settlement value to existing settled value if settlement date same as last settlement', async () => {
```
Comment (johnwatson484) [inline test/integration/local/settlement.test.js:88]: Duplicate test

File: test/unit/processing/confirm-due-dates.test.js [hunk 2]
```diff
@@ -110,37 +114,37 @@ describe('confirm due dates', () => {
...
       settledValue: 1000
     }]
     const confirmedPaymentRequests = confirmDueDates(paymentRequests, previousPaymentRequests, currentDate)
-    expect(confirmedPaymentRequests[0].dueDate).toBe('09/11/2021')
-    expect(confirmedPaymentRequests[0].schedule).toBe('M4')
+    expect(confirmedPaymentRequests[0].dueDate).toBe('09/11/2020')
+    expect(confirmedPaymentRequests[0].schedule).toBe('M12')
   })
 
   test('should update schedule to cover remaining payments when monthly and three remaining', () => {
```
Comment (marctemp) [inline test/unit/processing/confirm-due-dates.test.js:134]: Test case when monthly and 11 remaining (extreme edge case) -- can do all inclusive as you have Q4 and T4 but not required

File: test/unit/processing/confirm-due-dates.test.js [hunk 3]
```diff
@@ -586,4 +590,25 @@ describe('confirm due dates', () => {
     expect(confirmedPaymentRequests[0].dueDate).toBe('10/06/2022')
     expect(confirmedPaymentRequests[0].schedule).toBe('Q2')
   })
+
+  test('should update schedule to cover remaining payments when three day quarters and two remaining with partial settlement', () => {
```
Comment (marctemp) [inline test/unit/processing/confirm-due-dates.test.js:594]: Add a test case which tests our rounding issue

File: app/processing/confirm-due-dates.js [hunk 4]
```diff
@@ -39,39 +39,36 @@ const getSettledValue = (previousPaymentRequests) => {
   return previousPaymentRequests.filter(x => x.ledger === AP).reduce((x, y) => x + (y.settledValue ?? 0), 0)
 }
 
-const getPaymentSchedule = (schedule, dueDate, settledValue, totalValue, currentDate = new Date()) => {
-  const startDate = moment(dueDate, 'DD/MM/YYYY')
-  const frequency = schedule.charAt(0)
-  const totalPayments = schedule.substring(1, schedule.length)
```
[the remark this answers is not in the document]
Comment (marctemp) [reply app/processing/confirm-due-dates.js:45]: Is this something we should record in the asyncapi or README?

[the remark this answers is not in the document]
Comment (johnwatson484) [reply app/processing/confirm-due-dates.js:45]: Yes agreed. Async API spec should include it.  Updated.  Ideally the other repo's specs should have the same comment but we can update those in time.  It's more relevant for this service.

File: test/unit/processing/confirm-due-dates.test.js (hunk 2 above)
Comment (johnwatson484) [reply test/unit/processing/confirm-due-dates.test.js:314]: Fair comment.  Originally that wasn't needed as we assumed the frequency was derived.  Now we know it's not and we know all permutations makes sense to cover them with tests.

File: test/unit/processing/confirm-due-dates.test.js (hunk 3 above)
Comment (johnwatson484) [reply test/unit/processing/confirm-due-dates.test.js:774]: Yes agreed.  Originally I named this based on the assumed defect cause and neglected to change it.  Fixed now.

**Review by marctemp — APPROVED** [review]
All changes implemented
